app/middleware/error-handler.js

```diff
@@ -0,0 +1,47 @@
+'use strict'
+
+const logger = require('../utils/logger')(__filename)
+const { keys } = require('@govuk-pay/pay-js-commons').logging
+const {
+  NotAuthenticatedError,
+  UserAccountDisabledError,
+  NotAuthorisedError,
+  NotFoundError
+} = require('../errors')
+const paths = require('../paths')
+const { CORRELATION_HEADER } = require('../utils/correlation-header')
+const { renderErrorView } = require('../utils/response')
+
+module.exports = function errorHandler (err, req, res, next) {
+  const logContext = {}
+  logContext[keys.CORRELATION_ID] = req.headers[CORRELATION_HEADER]
+
+  if (res.headersSent) {
+    return next(err)
+  }
+
+  if (err instanceof NotAuthenticatedError) {
+    req.session.last_url = req.originalUrl
+    logger.info(`Redirecting attempt to access ${req.originalUrl} to ${paths.user.logIn}`, logContext)
+    return res.redirect(paths.user.logIn)
+  }
+
+  if (err instanceof UserAccountDisabledError) {
+    res.status(401)
+    return res.render('login/noaccess')
+  }
+
+  if (err instanceof NotAuthorisedError) {
+    return renderErrorView(req, res, 'You do not have the administrator rights to perform this operation.', 403)
+  }
+
+  if (err instanceof NotFoundError) {
+    res.status(404)
+    return res.render('404')
+  }
+
+  // trim the stack as long log messages are truncated by Splunk
+  logContext.stack = err.stack.substring(0, 300)
```

Comment:
Is this still true? I know @heathd did some digging into why Splunk/ the logging pipeline was being truncated. This may no longer be the case

Reply:
Aha https://payments-platform.atlassian.net/browse/PP-7003
Nice, I'll remove this